### ISSUE TITLE
 Eager macro-expansion failure: (wrong-number-of-arguments (1 . 1) 0) when (require 'dap-ui) #120 

### ISSUE THREAD

**Issue by zw963** [emacs-lsp/dap-mode#673]
This is not a error, just a warning when start emacs daemon.

Thanks

**Comment by yyoncho** [comment]
Please test with `M-x lsp-start-plain`

**Comment by jcs090218** [comment]
Looks similar to https://github.com/emacs-lsp/lsp-metals/issues/81. I think the issue is from the upstream `treemacs`.

I have reported the issue in https://github.com/Alexander-Miller/treemacs/issues/982.

**Comment by zw963** [comment]
I can reproduce use following minimum:

```cl
(require 'dap-mouse)
(require 'dapui)
```

Both of them required to output this error, although, treemacs really in the load path, it many be load automatically.

**Comment by zw963** [comment]
I can't reproduce it if `(require 'treemacs)` and `(require 'lsp-treemacs)`.

**Comment by jcs090218** [comment]
Do you have `lsp-mentals` installed? I think it get triggered only when you activated/executed.

**Comment by zw963** [comment]
> Do you have `lsp-mentals` installed? I think it get triggered only when you activated/executed.

No, i use newest master of lsp-mode and dap-mode, none of them exists `lsp-mentals`.

**Comment by jcs090218** [comment]
> I can't reproduce it if (require 'treemacs) and (require 'lsp-treemacs).

I don't think requiring it will cause the `Eager macro-expansion failure` error.

> No, i use newest master of lsp-mode and dap-mode, none of them exists lsp-mentals.

The error should came from this line 

https://github.com/emacs-lsp/dap-mode/blob/2cb49bb2ec22a7d6d4fd403bd4e2cc468f512501/dapui.el#L150

It's the same bug I have posted in the previous https://github.com/emacs-lsp/dap-mode/issues/673#issuecomment-1270377793.

**Comment by bertulli** [comment]
Please, forgive me if I missed something, but shouldn't the code of `dapui-loaded-sources` the one being fixed? For what I see, `treemacs` changed the interface to the `treemacs-initialize` function: can't `dap-mode`'s code be modified to reflect this?

**Comment by yyoncho** [comment]
@bertulli yes, it should. I will comment it out for now.

**Comment by yyoncho** [comment]
I just commented the source of dapui.el. @bertulli it is not that simple, we should rework the extension code to use lsp-treemacs-generic it is structured in different way.

**Comment by bertulli** [comment]
Got it, thank you! 🙏
